src/control/cmd/dmg/pretty/pool.go

```diff
@@ -172,14 +176,34 @@ func poolListCreateRow(pool *control.Pool) txtfmt.TableRow {
 			imbalance = pool.Usage[ti].Imbalance
 		}
 	}
+	var row txtfmt.TableRow
 
-	row := txtfmt.TableRow{
-		"Pool":      pool.GetName(),
-		"Size":      fmt.Sprintf("%s", humanize.Bytes(size)),
-		"State":     pool.State,
-		"Used":      fmt.Sprintf("%d%%", used),
-		"Imbalance": fmt.Sprintf("%d%%", imbalance),
-		"Disabled":  fmt.Sprintf("%d/%d", pool.TargetsDisabled, pool.TargetsTotal),
+	if upgrade == true {
+		var upgradeString string
+
+		if pool.PoolLayoutVer != pool.UpgradeLayoutVer {
+			upgradeString = fmt.Sprintf("%d->%d", pool.PoolLayoutVer, pool.UpgradeLayoutVer)
+		} else {
+			upgradeString = "None"
+		}
+		row = txtfmt.TableRow{
+			"Pool":      pool.GetName(),
+			"Size":      fmt.Sprintf("%s", humanize.Bytes(size)),
+			"State":     pool.State,
+			"Used":      fmt.Sprintf("%d%%", used),
+			"Imbalance": fmt.Sprintf("%d%%", imbalance),
+			"Disabled":  fmt.Sprintf("%d/%d", pool.TargetsDisabled, pool.TargetsTotal),
+			"Upgrade?":  upgradeString,
+		}
+	} else {
+		row = txtfmt.TableRow{
+			"Pool":      pool.GetName(),
+			"Size":      fmt.Sprintf("%s", humanize.Bytes(size)),
+			"State":     pool.State,
+			"Used":      fmt.Sprintf("%d%%", used),
+			"Imbalance": fmt.Sprintf("%d%%", imbalance),
+			"Disabled":  fmt.Sprintf("%d/%d", pool.TargetsDisabled, pool.TargetsTotal),
+		}
```

Comment:
No need to duplicate. A `TableRow` is just a map, so you can create the base row and then conditionally insert the upgrade cell data.

```suggestion
	row := txtfmt.TableRow{
		"Pool":      pool.GetName(),
		"Size":      fmt.Sprintf("%s", humanize.Bytes(size)),
		"State":     pool.State,
		"Used":      fmt.Sprintf("%d%%", used),
		"Imbalance": fmt.Sprintf("%d%%", imbalance),
		"Disabled":  fmt.Sprintf("%d/%d", pool.TargetsDisabled, pool.TargetsTotal),
	}

	if upgrade {
		upgradeString := "None"
		if pool.PoolLayoutVer != pool.UpgradeLayoutVer {
			upgradeString = fmt.Sprintf("%d->%d", pool.PoolLayoutVer, pool.UpgradeLayoutVer)
		}
		
		row["Upgrade?"] =  upgradeString
	}
```